### PR TITLE
#317 - added httpsRedirectCode

### PR DESCRIPTION
**Pull request by leostarcevic**

### CODE REVIEW

**Comment by vfarcic** [comment]
I did not yet have time to review it... Did you close it cause there's a problem with this PR or...?

**Comment by leostarcevic** [comment]
reviewing it myself I found a hardcoded value that needs to be changed, I'll reopen it when I'm done :)

File: proxy/template.go [hunk 1]
```diff
@@ -145,9 +145,12 @@ backend {{$.ServiceName}}-be{{.Port}}_{{.Index}}
 		{{- end}}
 		{{- if .DenyHttp}}
     http-request deny if !{ ssl_fc }
-		{{- end}}
-		{{- if .HttpsOnly}}
+        {{- end}}
+        {{- if .HttpsRedirectCode}}
+    redirect scheme https code {{.HttpsRedirectCode}} if !{ ssl_fc }
+		{{- else if .HttpsOnly}}
     redirect scheme https if !{ ssl_fc }
```
Comment (vfarcic) [inline proxy/template.go]: How about someting like:

```go
{{- if .HttpsRedirectCode}}
    redirect scheme https {{if .HttpsRedirectCode}}code {{.HttpsRedirectCode}}{{end}} if !{ ssl_fc }
{{- end}}
```

That would mean that `HttpsRedirectCode` is used only if `HttpsOnly`is `true`.

What do you think?

Comment (vfarcic) [reply proxy/template.go]: `if` was wrong. The snippet should be OK now.

Comment (leostarcevic) [reply proxy/template.go]: I agree 👍

**Comment by vfarcic** [comment]
I made a commit before merging this PR and that produced a few conflicts. The commits I made to this PR were only to resolve them. Also, I added an entry to the docs after merging this PR.

Thank you @leostarcevic.

**Pull request merged**